### ISSUE TITLE
current_line_blame errors on untracked files

### ISSUE THREAD

**Issue by MichaelRTalarico** [lewis6991/gitsigns.nvim#123]
**Describe the bug**
Receiving a s: expected string, got nil error upon startup

**To Reproduce**
Open file in Git directory (so far have tried Rust and C++ files)

lua file:

```vim
require("gitsigns").setup {
    signs = {
        add = {hl = "DiffAdd", text = "│", numhl = "GitSignsAddNr"},
        change = {hl = "DiffChange", text = "│", numhl = "GitSignsChangeNr"},
        delete = {hl = "DiffDelete", text = "_", numhl = "GitSignsDeleteNr"},
        topdelete = {hl = "DiffDelete", text = "‾", numhl = "GitSignsDeleteNr"},
        changedelete = {hl = "DiffChange", text = "~", numhl = "GitSignsChangeNr"}
    },
    numhl = false,
    keymaps = {
        -- Default keymap options
        noremap = true,
        buffer = true,
        ["n ]c"] = {expr = true, '&diff ? \']c\' : \'<cmd>lua require"gitsigns".next_hunk()<CR>\''},
        ["n [c"] = {expr = true, '&diff ? \'[c\' : \'<cmd>lua require"gitsigns".prev_hunk()<CR>\''},
        ["n <leader>gs"] = '<cmd>lua require"gitsigns".stage_hunk()<CR>',
        ["n <leader>gu"] = '<cmd>lua require"gitsigns".undo_stage_hunk()<CR>',
        ["n <leader>gr"] = '<cmd>lua require"gitsigns".reset_hunk()<CR>',
        ["n <leader>gp"] = '<cmd>lua require"gitsigns".preview_hunk()<CR>',
        ["n <leader>gb"] = '<cmd>lua require"gitsigns".blame_line()<CR>'
    },
    watch_index = {
        interval = 100
    },
    current_line_blame = true,
    sign_priority = 6,
    update_debounce = 100,
    status_formatter = nil, -- Use default
    use_decoration_api = true,
    use_internal_diff = true,  -- If luajit is prese
    debug_mode = true
}
```

1. Open code-based file in folder
2. Receive error immediately

Happens consistently every startup.

**Observed output**

> Error executing luv callback:
> vim/shared.lua:64: s: expected string, got nil
> stack traceback:
>         vim/shared.lua:64: in function 'gsplit'
>         vim/shared.lua:111: in function 'split'
>         ...ite/pack/packer/start/gitsigns.nvim/lua/gitsigns/git.lua:198: in function
>  '_user_on_exit'
>         .../site/pack/packer/start/plenary.nvim/lua/plenary/job.lua:237: in function
>  '_shutdown'
>         .../site/pack/packer/start/plenary.nvim/lua/plenary/job.lua:32: in function
> <.../site/pack/packer/start/plenary.nvim/lua/plenary/job.lua:23>

No additional output from debug mode.

**Screenshots**
<img width="819" alt="Screen Shot 2021-04-04 at 08 34 41" src="https://user-images.githubusercontent.com/2540878/113508909-a97d8600-9520-11eb-8fd7-fbe0312d582d.png">

**Additional context**
System: Mac
Neovim version: NVIM v0.5.0-dev+804ea22

**Comment by MichaelRTalarico** [comment]
Oh sorry, almost forgot, I have this included in my packer.vim

```vim
 use "nvim-lua/plenary.nvim"
 use "lewis6991/gitsigns.nvim"
```

**Comment by lewis6991** [comment]
There isn't really any specific instructions on how to reproduce this issue other than to open a file. Many people use this plugin and there is also a test suite that checks every commit so there must be something specific that's causing this otherwise the plugin is completely broken.

Is there anything else you can think off which would explain what's causing this? It's hard to help if I can't reproduce the issue.

**Comment by JonasUJ** [comment]
I came here because I also have this problem.

Steps to reproduce:
1. `:e new_file`
2. `:w`
3. Error

It's probably because the file is untracked.

**Comment by lewis6991** [comment]
Have you tried updating neovim to latest?

**Comment by lewis6991** [comment]
Ok, my apoligies. I've just realised this error is due to `config.current_line_blame`. With this turned off you do not get the error. 

I will update the issue template to instruct a minimum config should be supplied.

**Comment by JonasUJ** [comment]
Oh, I was just trying to build the latest but was having problems.

So does this mean current_line_blame shouldn't be used any more?

**Comment by lewis6991** [comment]
No, keep using it. It's a trivial issue, just fixing it now.

90% of the work of fixing issues is just figuring out how to reproduce and identify.

**Comment by JonasUJ** [comment]
Ok, that's great. Thanks for the quick answers :)

In case it's useful, here's some tracebacks:

When booting and moving the cursor:
```
Error executing luv callback:
vim/shared.lua:64: s: expected string, got nil
stack traceback:
        vim/shared.lua:64: in function 'gsplit'
        vim/shared.lua:111: in function 'split'
        ...ite/pack/packer/start/gitsigns.nvim/lua/gitsigns/git.lua:198: in function '_user_on_exit'
        .../site/pack/packer/start/plenary.nvim/lua/plenary/job.lua:237: in function '_shutdown'
        .../site/pack/packer/start/plenary.nvim/lua/plenary/job.lua:32: in function <.../site/pack/packer/start/plenary.
nvim/lua/plenary/job.lua:23>
        [C]: in function 'system'
        ...te/pack/packer/start/nvim-tree.lua/lua/nvim-tree/git.lua:11: in function 'update_root_status'
        ...te/pack/packer/start/nvim-tree.lua/lua/nvim-tree/git.lua:40: in function 'reload_roots'
        ...te/pack/packer/start/nvim-tree.lua/lua/nvim-tree/lib.lua:165: in function <...te/pack/packer/start/nvim-tree.
lua/lua/nvim-tree/lib.lua:159>
```
I noticed nvim-tree in the error, so uninstalled it and got this instead:
```
Error executing luv callback:
vim/shared.lua:64: s: expected string, got nil
stack traceback:
        vim/shared.lua:64: in function 'gsplit'
        vim/shared.lua:111: in function 'split'
        ...ite/pack/packer/start/gitsigns.nvim/lua/gitsigns/git.lua:198: in function '_user_on_exit'
        .../site/pack/packer/start/plenary.nvim/lua/plenary/job.lua:237: in function '_shutdown'
        .../site/pack/packer/start/plenary.nvim/lua/plenary/job.lua:32: in function <.../site/pack/packer/start/plenary.
nvim/lua/plenary/job.lua:23>
```

**Comment by lewis6991** [comment]
Not sure what the `nvim-tree` error is all about. I guess the error from gitsigns is interrpting it maybe?

Anyway I've fixed the issue in #124
